framework/test/mocha/unit/modules/chain/submodules/blocks/process.js

```diff
@@ -1596,165 +1598,419 @@ describe('blocks/process', () => {
+							const fn = Promise.promisify(cb);
+							fn().then(() => {
+								done();
+							});
+						};
+						blocksProcessModule.receiveBlockFromNetwork({
+							id: 5,
+							previousBlock: '3',
+							height: 3,
+						});
 					});
-					return expect(__private.receiveForkOne.args[0][2]).to.be.an(
-						'function'
-					);
 				});
 
-				it('should call __private.receiveForkOne', done => {
-					library.sequence.add = function(cb) {
-						const fn = Promise.promisify(cb);
-						fn().then(() => {
-							done();
+				describe('when block.previousBlock === lastBlock.previousBlock && block.height === lastBlock.height && block.id !== lastBlock.id', () => {
+					beforeEach(() => {
+						__private.receiveForkFive = sinonSandbox
+							.stub()
+							.callsArgWith(2, null, true);
+						return modules.blocks.lastBlock.get.returns({
+							id: '2',
+							height: 2,
+							previousBlock: '1',
 						});
-					};
-					blocksProcessModule.receiveBlockFromNetwork({
-						id: 5,
-						previousBlock: '3',
-						height: 3,
 					});
-				});
-			});
 
-			describe('when block.previousBlock === lastBlock.previousBlock && block.height === lastBlock.height && block.id !== lastBlock.id', () => {
-				beforeEach(() => {
-					__private.receiveForkFive = sinonSandbox
-						.stub()
-						.callsArgWith(2, null, true);
-					return modules.blocks.lastBlock.get.returns({
-						id: '2',
-						height: 2,
-						previousBlock: '1',
+					afterEach(() => {
+						expect(__private.receiveForkFive.calledOnce).to.be.true;
+						expect(__private.receiveForkFive.args[0][0]).to.deep.equal({
+							id: 5,
+							previousBlock: '1',
+							height: 2,
+						});
+						expect(__private.receiveForkFive.args[0][1]).to.deep.equal({
+							id: '2',
+							previousBlock: '1',
+							height: 2,
+						});
+						return expect(__private.receiveForkFive.args[0][2]).to.be.an(
+							'function'
+						);
+					});
+
+					it('should call __private.receiveForkFive', done => {
+						library.sequence.add = function(cb) {
+							const fn = Promise.promisify(cb);
+							fn().then(() => {
+								done();
+							});
+						};
+						blocksProcessModule.receiveBlockFromNetwork({
+							id: 5,
+							previousBlock: '1',
+							height: 2,
+						});
 					});
 				});
 
-				afterEach(() => {
-					expect(__private.receiveForkFive.calledOnce).to.be.true;
-					expect(__private.receiveForkFive.args[0][0]).to.deep.equal({
-						id: 5,
-						previousBlock: '1',
-						height: 2,
+				describe('when block.id === lastBlock.id', () => {
+					afterEach(done => {
+						expect(loggerStub.debug.args[0][0]).to.equal(
+							'Block already processed'
+						);
+						expect(loggerStub.debug.args[0][1]).to.equal('2');
+						done();
 					});
-					expect(__private.receiveForkFive.args[0][1]).to.deep.equal({
-						id: '2',
-						previousBlock: '1',
-						height: 2,
+
+					it('should log debug message and call a callback', done => {
```

Comment:
Is it possible you are not checking the log debug message for this test?

Reply:
Indeed. I am currently improving the tests so I will push the changes again shortly.